polaris-react/src/components/IndexTable/IndexTable.scss

```
@@ -403,10 +471,18 @@ $loading-panel-height: 53px;
   text-align: left;
   padding: var(--p-space-2) var(--p-space-4);
   white-space: nowrap;
+
+  #{$se23} & {
+    padding: var(--p-space-1_5-experimental);
+  }
```

Same suggestion to apply 16px right-side padding on the last TableCell in a TableRow as above for the last TableHeading

```suggestion

    &:last-child {
      padding-right: var(--p-space-3);
    }
  }
```